test/contracts-network/colony.js

```diff
@@ -419,22 +430,87 @@ contract("Colony", (accounts) => {
-      const colonyResolver = await Resolver.at(colonyResolverAddress);
+      const oldFunctions = new ethers.utils.Interface([
+        "function makeTask(uint256,uint256,bytes32,uint256,uint256,uint256)",
+        "function addPayment(uint256, uint256, address, address, uint256, uint256, uint256)",
+      ]);
+      const formattedOldABI = JSON.parse(oldFunctions.format(ethers.utils.FormatTypes.json));
+      console.log(JSON.stringify(formattedOldABI));
+
+      for (let i = 0; i < formattedOldABI.length; i += 1) {
+        for (let j = 0; j < formattedOldABI[i].inputs.length; j += 1) {
+          formattedOldABI[i].inputs[j].internalType = formattedOldABI[i].inputs[j].type;
+          formattedOldABI[i].inputs[j].name = "";
+        }
+      }
+
+      OldInterface = truffleContract({
+        abi: formattedOldABI,
+      });
 
-      const tasksPaymentsContract = await TasksPayments.new();
-      await colonyResolver.register("makeTask(uint256,uint256,bytes32,uint256,uint256,uint256)", tasksPaymentsContract.address);
-      await colonyResolver.register("addPayment(uint256,uint256,address,address,uint256,uint256,uint256)", tasksPaymentsContract.address);
-      tasksPayments = await TasksPayments.at(colony.address);
+      OldInterface.setProvider(web3.currentProvider);
     });
 
-    it("should be able to query for a task", async () => {
-      await tasksPayments.makeTask(1, UINT256_MAX, SPECIFICATION_HASH, 1, GLOBAL_SKILL_ID, 0, { from: USER0 });
+    beforeEach(async () => {
+      await colonyNetwork.createColony(token.address, 13, "", "");
+
+      const colonyIdx = await colonyNetwork.getColonyCount();
+      const colonyAddress = await colonyNetwork.getColony(colonyIdx);
+      colony = await IColony.at(colonyAddress);
+
+      const tokenLockingAddress = await colonyNetwork.getTokenLocking();
+      const tokenAuthority = await TokenAuthority.new(token.address, colony.address, [tokenLockingAddress]);
+      await token.setAuthority(tokenAuthority.address);
+
+      // Add old functions to the authority
+      const colonyAuthorityAddress = await colony.authority();
+      const authority = await ColonyAuthority.at(colonyAuthorityAddress);
+
+      const ADMINISTRATION_ROLE = 6;
+      const action1 = await encodeTxData(authority, "setRoleCapability", [
+        ADMINISTRATION_ROLE,
+        colony.address,
+        web3.utils.soliditySha3("makeTask(uint256,uint256,bytes32,uint256,uint256,uint256)").slice(0, 10),
+        true,
+      ]);
+      const action2 = await encodeTxData(authority, "setRoleCapability", [
+        ADMINISTRATION_ROLE,
+        colony.address,
+        web3.utils.soliditySha3("addPayment(uint256,uint256,address,address,uint256,uint256,uint256)").slice(0, 10),
+        true,
+      ]);
+
+      await colony.makeArbitraryTransactions([colonyAuthorityAddress, colonyAuthorityAddress], [action1, action2], true);
+
+      oldColony = await OldInterface.at(colony.address);
```

Comment:
Ditto all of this stuff, should be in a function in `scripts/deployOldUpgradeableVersion.js`

Reply:
Thinking about this some more, I've realised we can't actually remove old functions from `ColonyAuthority`, otherwise we break new deployments of old version of colonies (which, fundamentally, is what this extra setup was fixing). This is something that we should fix in the future for gas reasons, but for now I think we just restore the old functions in ColonyAuthority. Note that this will apply to #1173 as well.